### PR TITLE
Run solc-js tests on emscripten build

### LINKED ISSUE #2456: Travis Emscripten build should run the tests of solc-js

### PR DESCRIPTION
**Pull request by axic**
Fixes #2456.

### CODE REVIEW

**Review by chriseth — APPROVED** [review]

**Comment by axic** [comment]
@chriseth I think this should work now (waiting for travis), can you please review?

**Comment by chriseth** [comment]
Tests fail (it seems the new ones, too).

File: scripts/test_emscripten.sh [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+cd solc-js
+npm install
+
+# Replace soljson with current build
+echo "Replacing soljson.js"
+rm -f soljson.js
+# Make a copy because paths might not be absolute
+cp $REPO_ROOT/build/solc/soljson.js soljson.js
+
+# Update version (needed for some tests)
```
Comment (chriseth) [inline scripts/test_emscripten.sh]: Wouldn't it be better to supply an option to `npm run test` to ignore version errors?

Comment (axic) [reply scripts/test_emscripten.sh]: One option, but on the other hand, the requirement is that package.json matches the version of soljson.

In the future it should be able to handle multiple versions and we can remove this from here.

Comment (chriseth) [reply scripts/test_emscripten.sh]: This is a safety feature to be run before `npm publish`.

**Comment by axic** [comment]
@chriseth make sure to review this latest version, as we improve/change solcjs this can be improved/changed.

File: scripts/test_emscripten.sh [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+rm -f soljson.js
+# Make a copy because paths might not be absolute
+cp ../solc/soljson.js soljson.js
+
+# Update version (needed for some tests)
+echo "Creating version.c"
+rm -f version version.c
+cat <<EOF > version.c
+#include <stdio.h>
+#include "../include/solidity/BuildInfo.h"
```
Comment (chriseth) [inline scripts/test_emscripten.sh:52]: It is probably much easier to just read the version from the `CMakeLists.txt` - there should already be a routine in this directory to do this in the bash.

**Review by chriseth — APPROVED** [review]

**Pull request merged**